sov-modules/sov-modules-impl/src/example_simple_module/query.rs

```diff
@@ -0,0 +1,43 @@
+use super::ValueAdderModule;
+use serde::{Deserialize, Serialize};
+use sov_modules_api::QueryResponse;
+use sovereign_sdk::serial::{Decode, DecodeBorrowed};
+
+pub enum QueryMessage {
+    GetValue,
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq)]
+pub struct Response {
+    pub value: Option<u32>,
+}
+
+impl<C: sov_modules_api::Context> ValueAdderModule<C> {
+    /// Queries the state of the module.
+    pub fn query_value(&self) -> QueryResponse {
+        let response = Response {
+            value: self.value.get(),
+        };
+
+        let response = serde_json::to_vec(&response).unwrap();
```

Comment:
On second thought, maybe serialization should happen outside of the module system. If responses are always serialized, it will be hard to implement queries in higher-level modules. 

```
// token_module.rs
pub fn query_balance(&self, msg: address) -> QueryResponse { 
   let acct: Account = serde_json::deserialize(self.accounts_module.query(address)).unwrap();
   QueryResponse { balance: acct.balance }
 }

Reply:
Good point, I think methods that return `QueryResponse ` should be callable only from the outside. So in the example above `accounts_module` would have 2 methods:
- `get_balance->Balance` 
- `query-> QueryResponse ` which calls `get_balance` internally + adds serialization

The `token_module.rs` would call `get_balance` instead of `query`.
WDT?

Reply:
Yeah, that seems reasonable - but it would be nice to use the type system to show module designers what the correct pattern is as well. Let me think about this a bit more :)

Reply:
Ok, after thinking a bit more I don't know that using the type system to show the design pattern is worth all of the complexity. Let's stick with this for now.